conu/backend/docker/image.py

```diff
@@ -8,64 +12,94 @@
 
 class DockerImage(Image):
     """
-    A class which represents a docker container image. It contains utility methods to manipulate it.
+    Utility functions for docker images.
     """
     def __init__(self, repository, tag=None):
         """
         :param repository: str, image name, examples: "fedora", "registry.fedoraproject.org/fedora",
                             "tomastomecek/sen", "docker.io/tomastomecek/sen"
         :param tag: str, tag of the image, when not specified, "latest" is implied
         """
-        self.tag = tag or "latest"
-        self.name = repository
-        self.inspect_data = None
-        self.additional_names = []
+        super(DockerImage, self).__init__(repository, tag=tag)
+        self.tag = self.tag or "latest"
+
+    def __repr__(self):
+        return "DockerImage(repository=%s, tag=%s)" % (self.name, self.tag)
+
+    def __str__(self):
+        return self.get_full_name()
+
+    def get_full_name(self):
+        """
+        Provide full, complete image name
 
-    def full_name(self):
+        :return: str
+        """
         return "%s:%s" % (self.name, self.tag)
 
+    def get_id(self):
+        """
+        get unique identifier of this image
+
+        :return: str
+        """
+        if self._id is None:
+            self._id = self.get_metadata(refresh=False)["Id"]
+        return self._id
+
     def pull(self):
-        run_cmd("docker image pull %s" % self.full_name())
+        """
+        pull this image
+
+        :return: None
+        """
+        run_cmd("docker image pull %s" % self.get_full_name())
 
     def tag_image(self, repository=None, tag=None):
         """
         Apply additional tags to the image or even add a new name
 
         :param repository: str, see constructor
         :param tag: str, see constructor
-        :return: str, the new name
+        :return: instance of DockerImage
         """
         if not (repository or tag):
             raise ValueError("You need to specify either repository or tag.")
-        t = repository or self.name
-        t = "%s:%s" % (t, "latest" if not tag else tag)
-        run_cmd("docker image tag %s %s" % (self.full_name(), t))
-        self.additional_names.append(t)
-        return t
+        r = repository or self.name
+        t = "latest" if not tag else tag
+        image_name = "%s:%s" % (r, t)
+        run_cmd("docker image tag %s %s" % (self.get_full_name(), image_name))
+        return DockerImage(r, tag=t)
 
-    def __repr__(self):
-        return "Image(repository=%s, tag=%s)" % (self.name, self.tag)
+    def inspect(self, refresh=False):
+        """
+        return cached metadata by default (a convenience method)
 
-    def __str__(self):
-        return self.full_name()
+        :param refresh: bool, update the metadata with up to date content
+        :return: dict
+        """
+        return self.get_metadata(refresh=refresh)
+
+    def get_metadata(self, refresh=False):
```

Comment:
Similar to my comment for containers, I'd suggest using `refresh=True` for a default.

Reply:
the thing is that metadata of images tend not to change, but changing it to True anyway